scalafix-cli/src/main/scala/scalafix/internal/v1/Args.scala

```diff
@@ -367,6 +375,17 @@ case class Args(
       .map(_.stripPrefix(flag))
   }
 
+  def targetrootFromScalacOptions(): Option[String] =
+    dialect match {
+      case ScalafixConfig.Scala2 => semanticdbOption("targetroot")
```

Comment:
`semanticdbOption()` is used in other places - are the other usages valid in scala3? maybe the scala2/scala3 different mappings should be abstracted inside it?

Reply:
you would prefer something like that ? 
  def semanticdbOption2(settingInScala2: String, settingInScala3: String): Option[String] = {
    if (scalaVersion.isEmpty || scalaVersion.startsWith("2")) {
      val flag = s"-P:semanticdb:$settingInScala2:"
      scalacOptions
        .filter(_.startsWith(flag))
        .lastOption
        .map(_.stripPrefix(flag))
    } else {
      scalacOptions
        .sliding(2)
        .collectFirst {
          case h :: last :: Nil if h == settingInScala3 => last
        }
    }
  }

Reply:
yes! that would force any client of `semanticdbOption` to specify the scala2/scala3 flags (or maybe there is a pattern to derive one from the other?)

Reply:
Not really!